question-servers/freeform.js

```diff
@@ -134,10 +135,14 @@ module.exports = {
         if (courseElementsCache[course.id] !== undefined) {
             return callback(null, courseElementsCache[course.id]);
         }
-        module.exports.loadElements(path.join(course.path, 'elements'), 'course', (err, elements) => {
+        const coursePath = chunks.getRuntimeDirectoryForCourse(course);
+        chunks.ensureChunksForCourse(course.id, {'type': 'elements'}, (err) => {
```

Comment:
I believe this will lead to a bug where a PL server won't have up-to-date course element files if the elements were previously cached. In other words, a server will *never* get new course chunks once it has an entry in `courseElementsCache`. Once we throw multiple servers into the mix, and especially with syncs happening on different machines than actually serve questions, the course element cache won't work in its current form. There's a couple potential solutions here, I think:

* Key the cache on course ID *and* Git hash - this will invalidate the cache when a sync occurs, because a new hash will be associated with the course.
* Eliminate the cache entirely. Very robust, less efficient. a.k.a. "caching is hard, let's not."

The former is my preferred strategy. Anyone have any other thoughts?

Reply:
@nicknytko apparently already fixed this in #2507

Reply:
I'm filing this as a separate issue to be resolved after deployment. The initial courses we'll use chunks for don't have course elements.